TAB2MXL/src/main/java/View/StringParserUtilityBass.java

```diff
@@ -40,24 +58,128 @@
 			for (int i = 0; i < splitLines.length; i++) { // splitlines.length = how many lines there are
 				String currString = splitLines[i][j];
 				measure += currString;
-				System.out.println(currString); //testing
 				measure += "\n"; //testing
 			}
-			System.out.println(""); //testing
 			measureArray[measureCount] = measure;
 			measureCount++;
 		}
-		System.out.println(Arrays.deepToString(measureArray));
+		//System.out.println(Arrays.deepToString(measureArray));
 		
 		//call measureParser
 		for (int i = 0; i < measureArray.length; i++) {
 			measureList.add(measureParser(measureArray[i]));
 			setChord(measureList.get(i).getNoteList());
 			measureList.get(i).measureNumber = i + 1;
+			measureList.get(i).setTimeSignature(Controller.beatType);
+			
+			Map<Integer, Integer> timeList = Controller.beatList;
+			if(timeList.containsKey(i+1)) {
+				measureList.get(i).setTimeSignature(timeList.get(i+1));
+			}
+		}
+		
+		// Set measure repeats, if any
+		if (has_repeats) {
+			fillMeasureRepeats(rawLines);
 		}
 		return measureList;
 	}
 
+	public static void fillMeasureRepeats(String[] lines) throws Exception {
```

Comment:
these are recycled methods, we can just have this inherit from stringparserutility and override the methods that change